specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/preview/2019-06-01-preview/containerregistry_build.json

```diff
@@ -626,6 +926,55 @@
         }
       }
     },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ContainerRegistry/registries/{registryName}/taskRuns/{taskRunName}/listDetails": {
```

Comment:
please model it via a $expand semantics on taskRuns resource type itself. If the customer passes this query param, you can expand the details etc. btw, what is the extra information which is not returned in the a regular GET on taskRuns. 
A new resource type should not be exposed for this use case.

Reply:
@ravbhatnagar `listDetails` Post API will return additional sensitive information (e, secret values) for the resource. It will allow us to set additional permssion requirement for this operation.

We already use the pattern in task resource. 

https://github.com/Azure/azure-rest-api-specs/blob/master/specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/stable/2019-04-01/containerregistry_build.json#L646

Action | Microsoft.ContainerRegistry/registries/tasks/listDetails/action | List all details of a task for a container registry.

Reply:
@northtyphoon agreed (using post for explicit rbac for secrets). But it is GET /listdetails. change it to post.

Reply:
Thanks. It's the place we missed.

Reply:
@ravbhatnagar Updated to use the post for listdetails. And for the CI failures, all about breaking changes. Nothing I can do to solve it. Could you check?

Reply:
@huanwu - the problem here is that this POST call (which was earlier a GET) is returning the entire taskRUn object back. If it were returning just the secrets then a POST call would have been perfect. In this case, a user who does not have /read permission on taskRuns but only has this action permission can still read the entire content of taskRun by doing this pOST call. 
can you return only the secrets in this API? and the other details continue to be returned as part of the GET?

Reply:
@sanjaiganesh - FYI

Reply:
@ravbhatnagar the api is designed to return the complete payload including secrets. If the user is granted with listDetails/action permission, they are supposed to get everything for taskrun resource. I also think the action name (taskruns/listDetails/action) is clear to state the scope of the action.

Reply:
@northtyphoon - how we intend the customer to use our product or create custom roles may be different from how the customer actually uses it. As these are two separate APIs with corresponding RBAC operation, the customer can have a custom role which just gives /listDetails/action permission without giving them taskRuns/read permission. This is similar to someone having restart permission on the VM but not having read permission. In such cases, RBAC will be violated. Now, this may not be that common, but whenever it happens, it breaks RBAC intent. Why do you want to return the entire payload on the list call. Its common practice to surface metadata other than secrets on GET and secrets via POST. like storage accounts do similar thing

Reply:
/cc: @mnltejaswini

Reply:
The taskrun is a complex object. It can quite a few "secret" properties (eg, key=secretvalue1, key2=secretvalue2, arg1=secretarg1 and so on). It's not like storage case which is just the single key secret. From user's perspective, they need to inspect the secrets in the context of complete object payload. There is no much value to provide the seperate api to only get the secret values.

Again, we already used the pattern in tasks/listDetails and trained the customers.

Reply:
@ravbhatnagar another idea is can we do linked resource check to enforce the read access?